components/servo/lib.rs

```diff
@@ -926,3 +929,29 @@ fn create_sandbox() {
 fn create_sandbox() {
     panic!("Sandboxing is not supported on Windows, iOS, ARM targets and android.");
 }
+
+/// Implements GLContextDispatcher to dispatch functions from GLContext threads to the main thread's event loop.
+/// It's used in Windows to allow WGL GLContext sharing.
+pub struct MainThreadDispatcher {
+    compositor_proxy: Arc<Mutex<CompositorProxy>>,
```

Comment:
Do you need the arc and mutex here? `CompositorProxy` itself is `Send` and cloneable, and it seems the outer struct is cloned and moved around rather than shared around...

Reply:
Good point!